bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -84,38 +79,50 @@ enum NextEventResponse {
     },
 }
 
-fn next_event(ext_id: &str) -> Result<NextEventResponse> {
+async fn next_event(client: &reqwest::Client, ext_id: &str) -> Result<NextEventResponse> {
     let base_url = base_url(EXTENSION_ROUTE)
         .map_err(|e| Error::new(std::io::ErrorKind::InvalidData, e.to_string()))?;
     let url = format!("{base_url}/event/next");
-    ureq::get(&url)
-        .set(EXTENSION_ID_HEADER, ext_id)
-        .call()
+    client
+        .get(&url)
+        .header(EXTENSION_ID_HEADER, ext_id)
+        .send()
+        .await
         .map_err(|e| Error::new(std::io::ErrorKind::InvalidData, e.to_string()))?
-        .into_json()
+        .json()
+        .await
         .map_err(|e| Error::new(std::io::ErrorKind::InvalidData, e.to_string()))
 }
 
-fn register() -> Result<RegisterResponse> {
+async fn register(client: &reqwest::Client) -> Result<RegisterResponse> {
     let mut map = HashMap::new();
     let base_url = base_url(EXTENSION_ROUTE)
         .map_err(|e| Error::new(std::io::ErrorKind::InvalidData, e.to_string()))?;
     map.insert("events", vec!["INVOKE", "SHUTDOWN"]);
     let url = format!("{base_url}/register");
 
-    let resp = ureq::post(&url)
-        .set(EXTENSION_NAME_HEADER, EXTENSION_NAME)
-        .set(EXTENSION_ACCEPT_FEATURE_HEADER, EXTENSION_FEATURES)
-        .send_json(ureq::json!(map))
+    let resp = client
+        .post(&url)
+        .header(EXTENSION_NAME_HEADER, EXTENSION_NAME)
+        .header(EXTENSION_ACCEPT_FEATURE_HEADER, EXTENSION_FEATURES)
+        .json(&map)
+        .send()
+        .await
         .map_err(|e| Error::new(std::io::ErrorKind::InvalidData, e.to_string()))?;
 
     assert!(resp.status() == 200, "Unable to register extension");
 
     let extension_id = resp
-        .header(EXTENSION_ID_HEADER)
-        .unwrap_or_default()
+        .headers()
```

Comment:
Should we try to avoid using `unwrap` and `expect` here?

Reply:
in the test?

Reply:
the unwrap or default from before was cold comfort - we can't continue if there's no extension ID header in the response. That would be a major failure, and we'd crash anyway. In the future we'll enter a no-op loop, but for now I think this is okay